pkg/odo/cli/component/list.go

```diff
@@ -72,19 +74,39 @@ func (lo *ListOptions) Run() (err error) {
 			fmt.Println(string(out))
 		} else {
 			w := tabwriter.NewWriter(os.Stdout, 5, 2, 3, ' ', tabwriter.TabIndent)
-			fmt.Fprintln(w, "NAME", "\t", "TYPE", "\t", "SOURCE", "\t", "STATE", "\t", "CONTEXT")
+			fmt.Fprintln(w, "APP", "\t", "NAME", "\t", "TYPE", "\t", "SOURCE", "\t", "STATE", "\t", "CONTEXT")
 			for _, file := range components.Items {
-				fmt.Fprintln(w, file.Name, "\t", file.Spec.Type, "\t", file.Spec.Source, "\t", file.Status.State, "\t", file.Status.Context)
+				fmt.Fprintln(w, file.Spec.App, "\t", file.Name, "\t", file.Spec.Type, "\t", file.Spec.Source, "\t", file.Status.State, "\t", file.Status.Context)
 
 			}
 			w.Flush()
 		}
 		return nil
 	}
+	var components component.ComponentList
 
-	components, err := component.List(lo.Client, lo.Application)
-	if err != nil {
-		return errors.Wrapf(err, "failed to fetch components list")
+	if lo.allFlag {
+		apps, err := application.List(lo.Client)
+		if err != nil {
+			return err
+		}
+
+		var componentList []component.Component
+
+		for _, app := range apps {
+			comps, err := component.List(lo.Client, app)
+			if err != nil {
+				return err
+			}
+			componentList = append(componentList, comps.Items...)
+		}
+		components = component.GetMachineReadableFormatForList(componentList)
+	} else {
+
+		components, err = component.List(lo.Client, lo.Application)
+		if err != nil {
+			return errors.Wrapf(err, "failed to fetch components list")
+		}
```

Comment:
No comments anywhere :( if you could add comments as to what is happening that'd be great.